### DISCUSSION TITLE
Getting ready for publication

### DISCUSSION THREAD

**Discussion by amercader** [ckan/ckan-docker#32] [General]
Some initial thoughts from @kowh-ai:

- have the (existing) ckan-docker repo for the "get CKAN up on Docker ie: vanilla" repo
- create a new repo (maybe 'ckan-docker-dev') for a Dev orientated approach
- get rid of the "patches/" for vanilla, but keep for the Dev repo
- Get rid of 2.8 and 2.7. We should only support Docker on CKAN 2.9 (or latest stable which will soon be 2.10)
- Get rid of nginx, just provide instructions on how to add it
- Use pre-built CKAN base images, eventually move them over to an official CKAN DockerHub repo with the correct tags
- No extensions but provide instructions on how to add to a Dockerfile
- no hard-coded container names, use environment variables

**Comment by kowh-ai** [comment]
This was from a previous discussion, adding here:

- Be limited in scope, and act as a base that users can extend to their own needs
- Be opinionated, and provide one way to do things
- Be automatically tested

**Comment by amercader** [comment]
@kowh-ai see my thoughts below:

> - have the (existing) ckan-docker repo for the "get CKAN up on Docker ie: vanilla" repo
> - create a new repo (maybe 'ckan-docker-dev') for a Dev orientated approach
Maybe if we want to steer most users towards the custom image repo we should keep the `ckan-docker` name for that, so we could have:

* `ckan/ckan-docker-base`: For the base images Dockerfiles (prod and dev) and related scripts
* `ckan/ckan-docker`: For the project-oriented image template (prod and dev)

All the other images should live in separate repos like [ckan-solr](https://github.com/ckan/ckan-solr) for the Solr images. The datapusher ones should be built from the own datapusher repo (I think there's a Dockerfile there as well). Not sure if we need custom postgres ones, but if we do, again they can go to a separate repo

> - get rid of the "patches/" for vanilla, but keep for the Dev repo

+1

> - Get rid of 2.8 and 2.7. We should only support Docker on CKAN 2.9 (or latest stable which will soon be 2.10)

Yes, agree in dropping 2.7 and 2.8 support. Going forward we should support the same versions that we support in CKAN core, so current plus the previous one (so 2.9 now, 2.10 and 2.9 once 2.10 it's out). So let's plan the repo layout for having multiple versions (okfn one could be one potential approach, but maybe there's a better one)


> - Get rid of nginx, just provide instructions on how to add it

Not sure we need to go that far, but if it's easy to document we can do it


> - Use pre-built CKAN base images, eventually move them over to an official CKAN DockerHub repo with the correct tags

Yes, that's definitely the goal

> - No extensions but provide instructions on how to add to a Dockerfile

+1. I'm thinking that maybe we need to add a container for the CKAN worker on the compose setup

> - no hard-coded container names, use environment variables

Not sure what you mean here, can you explain it a bit?

**Reply by kowh-ai** [reply]
> Not sure what you mean here, can you explain it a bit?

We have a had a few Issues and PR's regarding the "hardcoding" of container names in the compose file
![_ckan](https://user-images.githubusercontent.com/54408245/177967148-ab33e082-f6cd-4298-9e6d-f0739641424e.png)
![_db](https://user-images.githubusercontent.com/54408245/177967212-76dfa803-1ef2-4f28-ade7-7dbbb418576e.png)

**Reply by kowh-ai** [reply]
> * `ckan/ckan-docker-base`: For the base images Dockerfiles (prod and dev) and related scripts

@amercader - Could you please create the `ckan/ckan-docker-base` repo - thanks!

**Reply by amercader** [reply]
Done, you should have admin access to it

**Reply by ondics** [reply]
Container names are critical e.g. when deploying multiple CKAN stacks within one docker system. +1 for getting rid of explicitly setting `container_name:`

**Comment by kowh-ai** [comment]
Getting close @amercader  - Seems to work OK - will work more on the security aspects from tomorrow

Feel free to take a look around when you get a spare mo, and rearrange whatever takes your fancy

The TODO's.txt file can be used to see what I need to do & what's left to do

**Comment by amercader** [comment]
Thanks @kowh-ai, great to hear. I have many open fronts but will try to have a look.

BTW I saw this and thought it would be useful to implement some basic tests to ensure that CKAN runs, files are where they are expected, etc: https://github.com/NVIDIA/container-canary

**Reply by kowh-ai** [reply]
> BTW I saw this and thought it would be useful to implement some basic tests to ensure that CKAN runs, files are where they are expected, etc: https://github.com/NVIDIA/container-canary

Yep - I took a look at this the other day - I think we can adapt for some rudimentary tests...thanks!
